src/Lexer.js

```diff
@@ -136,14 +136,15 @@ module.exports = class Lexer {
       }
 
       // code
-      if (token = this.tokenizer.code(src, tokens)) {
+      if (token = this.tokenizer.code(src)) {
         src = src.substring(token.raw.length);
-        if (token.type) {
-          tokens.push(token);
-        } else {
-          lastToken = tokens[tokens.length - 1];
+        lastToken = tokens[tokens.length - 1];
+        // An indented code block cannot interrupt a paragraph.
+        if (lastToken && lastToken.type === 'paragraph') {
```

Comment:
I wonder if we should check for a paragraph before we call the `code` tokenizer? That might save some work that doesn't need to be done.

Reply:
Probably worth trying. It's a tradeoff of always checking `LastToken` and sometimes calling `codeTokenizer` vs always calling `codeTokenizer` and sometimes checking `LastToken`. I'm not sure how often this situation comes up that one would be better than the other.

Reply:
> If LastToken is a paragraph though what do we do? Just continue? or call the "text" tokenizer out of sequence?

Good point. I suppose we would still need to call code tokenizer to see if we should skip the other tokens. Maybe it is still better to check code tokenizer first.